src/PostgREST/DbRequestBuilder.hs

```diff
@@ -324,7 +324,11 @@ addProperty f (targetNodeName:remainingPath, a) (Node rn forest) =
 mutateRequest :: Schema -> TableName -> ApiRequest -> S.Set FieldName -> [FieldName] -> ReadRequest -> Either Response MutateRequest
 mutateRequest schema tName apiRequest cols pkCols readReq = mapLeft errorResponseFor $
   case action of
-    ActionCreate -> Right $ Insert qi cols ((,) <$> iPreferResolution apiRequest <*> Just pkCols) [] returnings
+    ActionCreate -> do
+        confCols <- case iOnConflict apiRequest of
+            Nothing -> pure pkCols
```

Comment:
The [error](https://circleci.com/gh/PostgREST/postgrest/3968?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) on CI it's related to alignment. Try:
```suggestion
            Nothing    -> pure pkCols
```